docs/src/operations.md

```diff
@@ -211,3 +211,102 @@ BinaryOperation at (Center, Center, Center)
+We start by creating a latitude-longitude grid that only goes up to 30 degrees latitude.
+Conveniently, with this latitude extent that grid covers half the total area of the sphere, i.e., `2π * grid.radius^2`.
+
+Let's try to estimate this area using `Integral` operation.
+We create a Field, we fill it with ones and we integrate it over the whole grid.
+
+
+```jldoctest operations_avg_int
+using Oceananigans
+
+grid = LatitudeLongitudeGrid(size=(60, 10, 5),
+                             latitude = (-30, 30),
+                             longitude = (0, 360),
+                             z = (-1000, 0))
+
+c = CenterField(grid)
+set!(c, 1)
+
+∫c = Field(Integral(c, dims=(1, 2)))
+
+# output
+1×1×5 Field{Nothing, Nothing, Center} reduced over dims = (1, 2) on LatitudeLongitudeGrid on CPU
+├── data: OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, size: (1, 1, 5)
+├── grid: 60×10×5 LatitudeLongitudeGrid{Float64, Periodic, Bounded, Bounded} on CPU with 3×3×3 halo and with precomputed metrics
+├── operand: Integral of BinaryOperation at (Center, Center, Center) over dims (1, 2)
+├── status: time=0.0
+└── data: 1×1×11 OffsetArray(::Array{Float64, 3}, 1:1, 1:1, -2:8) with eltype Float64 with indices 1:1×1:1×-2:8
+    └── max=0.0, min=0.0, mean=0.0
+```
+
+A few remarks: note that the `∫c` has locations `Nothing, Nothing, Center`; this is because we have integrated in the first two dimensions and thus it's `reduced over dims = (1, 2)`. 
+`c` was initialised as a `CenterField` but the above also applies to `XFaceField`, `YFaceField` and ZFaceField`. 
+Further note that `∫c` is full of zeros; its max, min, and mean values are all 0.
+No computation has been done yet.
+To compute `∫c`, we call [`compute!`](@ref),
+
+```jldoctest operations_avg_int
+compute!(∫c)
+
+# output
+1×1×5 Field{Nothing, Nothing, Center} reduced over dims = (1, 2) on LatitudeLongitudeGrid on CPU
+├── data: OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, size: (1, 1, 5)
+├── grid: 60×10×5 LatitudeLongitudeGrid{Float64, Periodic, Bounded, Bounded} on CPU with 3×3×3 halo and with precomputed metrics
+├── operand: Integral of BinaryOperation at (Center, Center, Center) over dims (1, 2)
+├── status: time=0.0
+└── data: 1×1×11 OffsetArray(::Array{Float64, 3}, 1:1, 1:1, -2:8) with eltype Float64 with indices 1:1×1:1×-2:8
+    └── max=2.55032e14, min=2.55032e14, mean=2.55032e14
+```
+
+Above we see that the max, min and mean of the field are all the same.
+Let's check that these values are what we expect:
+
+```jldoctest operations_avg_int
+∫c[1, 1, 1] ≈ 2π * grid.radius^2 # area of spherical zone with |φ| ≤ 30ᵒ
+
+# output
+
+true
+```
+
+We can further have conditional reduced operations. Let's compute the above integral but only for North hemisphere.
+
+First we need to define a condition which is a function with arguments `(i, j, k, grid, field)` that returns true or false.
```

Comment:
I think it might be useful (down the track) to talk about how to input your own Boolean masks. While functions are useful for simple slicing (e.g. Northern Hemisphere) if we want to isolate an irregular region (e.g., Gulf of Mexico, Mediterranean, Hudson Bay) it may be more useful to input a bespoke Boolean matrix. I am testing this out at the moment and can add a note when I have a better sense.

Reply:
We can definitely showcase here how an array can be provided for `condition`, is this what you mean?